core/os/file_access.cpp

```diff
@@ -273,9 +273,72 @@ String FileAccess::get_token() const {
 	return String::utf8(token.get_data());
 }
 
+class CharBuffer {
+	PoolVector<char> vector;
+	PoolVector<char>::Write vector_write;
+
+	char stack_buffer[256];
+
+	char *buffer;
+	int capacity;
+	int written;
+
+	bool grow() {
+
+		vector_write = PoolVector<char>::Write(); // unlock for resize
+
+		if (vector.resize(MAX(next_power_of_2(1 + written), vector.size() * 2)) != OK) {
+
+			return false;
+		}
+
+		vector_write = PoolVector<char>::Write(vector.write());
+
+		if (buffer == stack_buffer) { // first chunk?
+
+			for (int i = 0; i < written; i++) {
+
+				vector_write[i] = stack_buffer[i];
+			}
+		}
+
+		buffer = vector_write.ptr();
+		capacity = vector.size();
+
+		return true;
+	}
+
+public:
+	_FORCE_INLINE_ CharBuffer()
```

Comment:
Small style issue here (well your style is fine, but not consistent with that enforced by `clang-format` 5.0.0): https://travis-ci.org/godotengine/godot/jobs/316057663